### PR TITLE
feat: Support language override in UI

### PR DESCRIPTION
**Pull request by dcramer**

### CODE REVIEW

**Comment by dcramer** [comment]
@billyvg @lynnagara i cant figure out why this is causing test failures, let alone the failures its causing

```
 FAIL  tests/js/spec/views/settings/organizationProjects.spec.jsx
  ● OrganizationProjects › Should render the projects in the store

    expect(value).toMatchSnapshot()

    Received value does not match stored snapshot 1.

    - Snapshot
    + Received

    @@ -17,10 +17,11 @@
                  <Button
                    disabled={false}
                    icon="icon-circle-add"
                    priority="primary"
                    size="small"
    +               title={undefined}
                    to="/organizations/org-slug/projects/new/"
                  >
                    Create Project
                  </Button>
                }

      42 |       routerContext
      43 |     );
    > 44 |     expect(wrapper).toMatchSnapshot();
      45 |
      46 |     expect(wrapper.find('.project-name').text()).toBe('project-slug');
      47 |

      at Object.<anonymous> (tests/js/spec/views/settings/organizationProjects.spec.jsx:44:21)

 › 1 snapshot test failed.

 FAIL  tests/js/spec/views/onboarding/index.spec.jsx
  ● OnboardingWizard › render() › should render and respond to click events

    expect(value).toMatchSnapshot()

    Received value does not match stored snapshot 2.

    - Snapshot
    + Received

    @@ -101,16 +101,10 @@
                    "calls": Array [
                      Array [
                        "csharp",
                      ],
                    ],
    -               "results": Array [
    -                 Object {
    -                   "isThrow": false,
    -                   "value": undefined,
    -                 },
    -               ],
                  }
                }
                setTeam={[Function]}
                team=""
                teams={Array []}
    @@ -131,16 +125,10 @@
                        "calls": Array [
                          Array [
                            "csharp",
                          ],
                        ],
    -                   "results": Array [
    -                     Object {
    -                       "isThrow": false,
    -                       "value": undefined,
    -                     },
    -                   ],
                      }
                    }
                    setTeam={[Function]}
                    showOther={true}
                    team=""

      70 |       let node = wrapper.find('PlatformCard').first();
      71 |       node.simulate('click');
    > 72 |       expect(wrapper).toMatchSnapshot();
      73 |     });
      74 |   });
      75 | });

      at Object.<anonymous> (tests/js/spec/views/onboarding/index.spec.jsx:72:23)
```

**Comment by billyvg** [comment]
Your yarn  is outdated

File: src/sentry/static/sentry/app/stores/configStore.jsx [hunk 1]
```diff
@@ -22,15 +22,17 @@ const ConfigStore = Reflux.createStore({
...
+  loadInitialData(config, languageOverride = null) {
     config.features = new Set(config.features || []);
     this.config = config;
 
     // TODO(dcramer): abstract this out of ConfigStore
     if (config.user) {
       config.user.permissions = new Set(config.user.permissions);
       moment.tz.setDefault(config.user.options.timezone);
-      setLocale(config.user.options.language || 'en');
+      setLocale(languageOverride || config.user.options.language || 'en');
```
Comment (billyvg) [inline src/sentry/static/sentry/app/stores/configStore.jsx]: How does django determine the language code from `request.LANGUAGE_CODE`? Is there a language supplied when the `lang` query param is not set? Should user's option take precedence?

Comment (dcramer) [reply src/sentry/static/sentry/app/stores/configStore.jsx]: @billyvg uses the same option from the user via our middleware, so ultimately its going to be identical to config.user.options.language

**Review by billyvg — APPROVED** [review]

**Pull request merged**